java/ql/src/semmle/code/java/security/InformationLeak.qll

```diff
@@ -0,0 +1,23 @@
+/** Provides classes to reason about System Information Leak vulnerabilities. */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.ExternalFlow
+
+/** CSV sink models representing methods not susceptible to XSS but outputing to an HTTP response body. */
+private class DefaultInformationLeakSinkModel extends SinkModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "javax.servlet.http;HttpServletResponse;false;sendError;(int,String);;Argument[1];information-leak"
+      ]
+  }
+}
+
+/** A sink that represent a method that outputs data to an HTTP response. */
+abstract class InformationLeakSink extends DataFlow::Node { }
+
+/** A default sink representing methods outputing data to an HTTP response. */
+private class DefaultInformationLeakSink extends InformationLeakSink {
+  DefaultInformationLeakSink() { sinkNode(this, "information-leak") }
```

Comment:
If I understand correctly, an XSS sink is always an information-leak sink, right? In that case, we could add `or this instanceof XssSink` to `DefaultInformationLeakSink()` here instead of explicitly calling out `XssSink` in the information leak query.

Reply:
Makes sense, will move the check to `DefaultInformationLeakSink`